src/deploy/hosting/convertConfig.ts

```diff
@@ -136,8 +151,43 @@ export async function convertConfig(
       }
       const id = rewrite.function.functionId;
       const region = rewrite.function.region;
-      const endpoint = findEndpointForRewrite(deploy.config.site, haveBackend, id, region);
+      let endpoint: backend.Endpoint | undefined = undefined;
+      let functionIsInDeployingBackends = false;
```

Comment:
again, this seems like a place where backend.merge could have helped. What if above you merged not only the wantBackend but the exitingBackend as well?

The main reason I would argue against simplifying would be to set you up for more advanced error handling where you notice that a function currently exists but is about to get deleted.
1. I don't know if that would be necessary
2. You could still use some utility methods/comprehensions to do that. Off the top of my head:

```
wantBackend = backend.merge(...Object.keys(payload.functions || {}).map(c => c.wantBackend));
haveBackendTargets = backend.merge(...Object.keys(payload.functions || {}).map(c => c.haveBackend));
// Just putting this in a backend since your utility methods use them already
// You can use any predicate, but I'm going to rely on the structure of Backend to avoid N^2 performance
toDelete = backend.matchingBackend(haveBackendTargets, (e) => wantBackend.endpoints[e.region]?.[e.id]);
```

Reply:
I added a test case for function-being-deleted and fixed it. Turned out to be reasonably easy to do. I'm leaving haveBackend and wantBackend separate because it's much simpler to understand that way, especially for the function-being-deleted case.